src/macro_support.rs

```diff
@@ -0,0 +1,39 @@
+use core::cmp::Ordering;
+
+#[inline]
+fn flatten<T>(value: (usize, &Option<T>)) -> Option<(usize, &T)> {
+    match value {
+        (index, Some(value)) => Some((index, value)),
+        _ => None,
+    }
+}
+
+/// `partial_cmp` implementation over iterators which ensures that storage
+/// ordering between `None` and `Some` is handled in a reasonable manner.
+#[doc(hidden)]
+#[allow(clippy::missing_inline_in_public_items)]
```

Comment:
Why not mark these as `#[inline]`?

Reply:
The reasoning was that it's a big function body. It's probably better to let Rust decide what shouldn't and should be inlined with a slight preference towards code re-use to avoid bloat.

I plan to do some comparisons later both with speed and code size.

Reply:
It's a generic function, so implicitly inline-able anyways.

Reply:
Ah, so to my understanding generic functions aren't inherently more inline-able than regular functions. What I think will happen is that any calls to this function which has a compatible set of type parameters will re-use the same underlying implementation *unless* rustc decides that it's better to inline it.

E.g. 3 uses but two functions: https://godbolt.org/z/383PY65vW

This is just an instance where I think the generated function is *kinda* large. So I'd rather hint that it should use its regular heuristics rather than inline it to encourage code re-use.